internal/provider/resources/resource_cluster_workspace_authorization.go

```diff
@@ -0,0 +1,381 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/platform"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/models"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/schemas"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+var _ resource.Resource = &ClusterWorkspaceAuthorizationResource{}
+var _ resource.ResourceWithConfigure = &ClusterWorkspaceAuthorizationResource{}
```

Comment:
lets add the ResourceWithImport as well
it will be a little different but my team_roles PR #64 one should be similar

we should also add an acceptance test for it as well